src/SAML2/XML/saml/OneTimeUse.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace SAML2\XML\saml;
+
+use DOMElement;
+use SAML2\Constants;
+use Webmozart\Assert\Assert;
+
+final class OneTimeUse extends AbstractConditionType
```

Comment:
Having a look at the standard, I think we can completely drop this class and use instead a boolean in the classes referencing it. In the end, this element being present signals one thing, and the lack of it signals the opposite. Therefore, in `Conditions` we could just have an `isOneTimeUse()` method as a substitute (maybe even receiving a bool parameter to act both as getter and setter), and that would suffice for the use of this element we need to do. The only downside I see there is that then `Conditions::fromXML()` becomes a bit more complicated, since you need to test for the element, instead of just calling `getChildrenOfClass()`

Reply:
I went for consistency here, but I can change it if you like

Reply:
I think it would be simpler to use, because the way it is now you need to call the getter, and see if it's not `null`, which in my head is a very convoluted way to test a boolean condition. Also, having this as an independent class, means we need to test it as well...